### PR TITLE
Fix some warnings as if they were active

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/GameCommands.h [hunk 1]
```diff
@@ -343,7 +343,7 @@ namespace OpenLoco::GameCommands
             , rotation(regs.bh & 0x3)
             , trackId(regs.dl & 0x3F)
             , index(regs.dh)
-            , trackObjectId(regs.ebp)
+            , trackObjectId(static_cast<uint8_t>(regs.ebp)) // TODO: Validate this.
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/GameCommands.h:346]: This is safe

File: src/OpenLoco/src/GameCommands/GameCommands.h [hunk 2]
```diff
@@ -452,7 +452,7 @@ namespace OpenLoco::GameCommands
         RenameStationArgs() = default;
         explicit RenameStationArgs(const registers& regs)
             : stationId(StationId(regs.cx))
-            , nameBufferIndex(regs.ax)
+            , nameBufferIndex(static_cast<uint8_t>(regs.ax)) // TODO: Validate this.
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/GameCommands.h:455]: So is this

File: src/OpenLoco/src/GameCommands/GameCommands.h [hunk 3]
```diff
@@ -596,8 +596,8 @@ namespace OpenLoco::GameCommands
             , rotation(regs.bh & 0x3)
             , trackId(regs.dl & 0xF)
             , index(regs.dh & 0x3)
-            , trackObjectId(regs.bp)
-            , type(regs.edi >> 16)
+            , trackObjectId(static_cast<uint8_t>(regs.bp)) // TODO: Validate this.
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/GameCommands.h:599]: This is safe

File: src/OpenLoco/src/GameCommands/GameCommands.h [hunk 4]
```diff
@@ -1403,8 +1403,8 @@ namespace OpenLoco::GameCommands
             , rotation(regs.bh & 0x3)
             , roadId(regs.dl & 0xF)
             , index(regs.dh & 0x3)
-            , roadObjectId(regs.bp)
-            , type(regs.edi >> 16)
+            , roadObjectId(static_cast<uint8_t>(regs.bp)) // TODO: Validate this.
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/GameCommands.h:1406]: Safe

File: src/OpenLoco/src/GameCommands/GameCommands.h [hunk 5]
```diff
@@ -1943,7 +1943,7 @@ namespace OpenLoco::GameCommands
     {
         registers regs;
         regs.bl = Flags::apply;
-        regs.ax = ax;
+        regs.ax = static_cast<int16_t>(ax); // Should the argument be int16_t rather?
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/GameCommands.h:1946]: Value will always be very small number so could just make it uint8

File: src/OpenLoco/src/GameCommands/GameCommands.h (hunk 1 above)
Comment (ZehMatt) [reply src/OpenLoco/src/GameCommands/GameCommands.h:346]: I know this is safe but why use the 32 bit width of that register? That's why I added the validate comments all over as I'm not sure if thats the actual register used and we could avoid the casts if we just use the actual 8 bit version of it.

File: src/OpenLoco/src/GameCommands/GameCommands.h (hunk 2 above)
Comment (ZehMatt) [reply src/OpenLoco/src/GameCommands/GameCommands.h:455]: Shouldn't that be al then?

File: src/OpenLoco/src/GameCommands/GameCommands.h (hunk 3 above)
Comment (ZehMatt) [reply src/OpenLoco/src/GameCommands/GameCommands.h:599]: Shouldn't this be rather bl?

File: src/OpenLoco/src/GameCommands/GameCommands.h (hunk 4 above)
Comment (ZehMatt) [reply src/OpenLoco/src/GameCommands/GameCommands.h:1406]: Also probably bl?

File: src/OpenLoco/src/GameCommands/GameCommands.h (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/GameCommands/GameCommands.h:346]: There is no 8bit ebp register. Conversion from registers to values it doesn't matter much if the register is wider than it needs to be. The other way round though should be investigated to ensure its correct.

File: src/OpenLoco/src/GameCommands/GameCommands.h (hunk 3 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/GameCommands/GameCommands.h:599]: No there is no 8bit bp register

**Review by duncanspumpkin — APPROVED** [review]

**Review by AaronVanGeffen — APPROVED** [review]

**Comment by AaronVanGeffen** [comment]
There's a merge conflict due to GameCommands.h having been split up since this PR was opened. Could you rebase?

**Comment by AaronVanGeffen** [comment]
This PR is now quite outdated. Considering the changes to Speed.hpp have already been ported in #2138, I think it's best to close this PR. @ZehMatt, do you agree?

**Pull request closed without merging**